tests/client/amo/components/TestLink.js

```diff
@@ -4,119 +4,193 @@ import {
   renderIntoDocument,
 } from 'react-addons-test-utils';
 import { findDOMNode } from 'react-dom';
-import { Link } from 'react-router';
+import { Link as ReactRouterLink } from 'react-router';
 
-import { LinkBase, mapStateToProps } from 'amo/components/Link';
+import Link, { mapStateToProps } from 'amo/components/Link';
+import createStore from 'amo/store';
+import { setClientApp, setLang } from 'core/actions';
 
 
 describe('<Link />', () => {
-  const defaultState = { api: { clientApp: 'android', lang: 'fr' } };
+  const { store } = createStore();
```

Comment:
I think it's a lot safer to wrap this kind of thing in `beforeEach()`. That way you can guarantee that no dispatched actions from one test will leak into another.